sensor/common/detector/detector.go

```diff
@@ -165,6 +177,7 @@ type detectorImpl struct {
 
 	networkFlowsQueue *queue.Queue[*queue.FlowQueueItem]
 	indicatorsQueue   *queue.Queue[*queue.IndicatorQueueItem]
+	deploymentsQueue  queue.SimpleQueue[*queue.DeploymentQueueItem]
```

Comment:
The two other queues above are stored using reference `*queue...`. Do you maybe want to do it here as well?

Reply:
I used the `queue.Queue` wrapper in the first commit of this PR but I realized that it was confusing because the deployment queue does not need to be pause/resume and in order to make it work I needed to `Start` and `Resume` the queue. So:

- The other two queues need to use a wrapper `queue.Queue` to be able to pause and resume the pulling logic.
- But the deployment queue does not need to be pause/resume, so we can use the `pkg/queue` directly.

In this case `SimpleQueue` is an interface for the `pkg/queue`.